src/__tests__/EventValidator_test.ts

```diff
@@ -0,0 +1,1143 @@
+        props: {
+          prop_a: createPropertySpec({ id: "prop_a" }),
+          prop_b: createPropertySpec({ id: "prop_b" })
+        }
+      })
+    ];
+
+    const result = findClosestMatch("test_event", { prop_a: "a", prop_b: "b" }, events);
+
+    expect(result?.event.id).toBe("evt_2");
+  });
+
+  test("should score higher for enum value matches", () => {
+    const events = [
+      createEventSpec({
+        id: "evt_1",
+        name: "login",
+        props: {
+          method: createPropertySpec({
+            id: "prop_method",
+            v: ["email", "google"]
+          })
+        }
+      })
+    ];
+
+    const result = findClosestMatch("login", { method: "email" }, events);
+
+    expect(result).not.toBeNull();
+    // Internal: Verify score is computed (testing implementation detail)
+    expect(result?.score).toBeGreaterThan(0);
+  });
+
+  describe("Internal Scoring (implementation details)", () => {
+    test("should give bonus score for variant with mappedName", () => {
+      const events = [
+        createEventSpec({
+          id: "evt_1",
+          name: "test_event",
+          props: {},
+          variants: [
+            createVariantSpec({
+              variantId: "var_1",
+              eventId: "evt_1",
+              nameSuffix: "WithMapped",
+              mappedName: "Mapped Event Name",
+              props: {}
+            }),
+            createVariantSpec({
+              variantId: "var_2",
+              eventId: "evt_1",
+              nameSuffix: "WithoutMapped",
+              props: {}
+            })
+          ]
+        })
+      ];
+
+      // Match the event
+      const result = findClosestMatch("test_event", {}, events);
+      expect(result).not.toBeNull();
+
+      // The variant with mappedName should get a small bonus
+      // This tests the VARIANT_MAPPED_NAME scoring weight
+    });
+  });
+});
+
+// =============================================================================
+// VALIDATION LOGIC TESTS
+// =============================================================================
+
+describe("validateProperties", () => {
+  describe("RequiredMissing", () => {
+    test("should report missing required property", () => {
+      const event = createEventSpec({
+        props: {
+          required_prop: createPropertySpec({ id: "prop_req", r: true })
+        }
+      });
+
+      const errors = validateProperties({}, event, null);
+
+      expect(errors).toHaveLength(1);
+      expect(errors[0].code).toBe("RequiredMissing");
+      expect(errors[0].propertyId).toBe("prop_req");
+      expect(errors[0].propertyName).toBe("required_prop");
+    });
+
+    test("should not report missing optional property", () => {
+      const event = createEventSpec({
+        props: {
+          optional_prop: createPropertySpec({ r: false })
+        }
+      });
+
+      const errors = validateProperties({}, event, null);
+
+      expect(errors).toHaveLength(0);
```

Comment:
do we need to validate the schema errors, (missing/optional/wrong type), will the inspector not just take care of that later downstream?